File: golang-for-nodejs-developers/examples/files.js
import { openSync, writeSync, readSync, closeSync, unlinkSync } from 'node:fs'
import { Buffer } from 'node:buffer'

// create file
const createFd = openSync('test.txt', 'w')
closeSync(createFd)

// open file (returns file descriptor)
const fd = openSync('test.txt', 'r+')

const wbuf = Buffer.from('hello world.')
const rbuf = Buffer.alloc(12)
const off = 0
const len = 12
const pos = 0

// write file
writeSync(fd, wbuf, 0, wbuf.length, pos)

// read file
readSync(fd, rbuf, off, len, pos)

console.log(rbuf.toString())

// close file
closeSync(fd)

// delete file
unlinkSync('test.txt')

File: golang-for-nodejs-developers/examples/files.go
package main

import (
	"fmt"
	"os"
)

func main() {
	// create file
	file, err := os.Create("test.txt")
	if err != nil {
		panic(err)
	}

	// close file
	file.Close()

	// open file
	file, err = os.OpenFile("test.txt", os.O_RDWR, 0755)
	if err != nil {
		panic(err)
	}
	defer file.Close()

	wbuf := []byte("hello world.")
	rbuf := make([]byte, 12)
	var off int64

	// write file
	if _, err := file.WriteAt(wbuf, off); err != nil {
		panic(err)
	}

	// read file
	if _, err := file.ReadAt(rbuf, off); err != nil {
		panic(err)
	}

	fmt.Println(string(rbuf))

	// delete file
	if err := os.Remove("test.txt"); err != nil {
		panic(err)
	}
}
